Port Members: Create `getMembersForColony' lambda function & support Members aside on Colony home

danbr opened the pull request:
## Description

This PR now covers 2 issues.

1. 'getMembersForColony' lambda function (This was originally implemented in #75 - but work continued while working on Members aside)
2. Update colony home members/watchers aside to correctly show both contributors and watchers.


<img width="493" alt="Screenshot 2022-12-02 at 18 24 06" src="https://user-images.githubusercontent.com/582700/205310604-b2ad04e5-5bc6-4887-9cf3-8c07420a075c.png">


Resolves #68 & #86

jakubcolony reviewed (commented):
@danbr I'm seeing this:
<img width="224" alt="image" src="https://user-images.githubusercontent.com/112586815/205960687-1fbe02e9-8e17-4f3b-a184-a0ff3befecb4.png">

Together with errors from amplify container:
<img width="756" alt="image" src="https://user-images.githubusercontent.com/112586815/205960863-206de1bc-a9b6-4832-96db-1699f6af3f30.png">

Am I doing something wrong?

(At least we know error handling is working 😉)

From what I was able to test, loading looks a bit ugly (I guess not a big deal since we've got a UI refresh coming): 
<img width="203" alt="Screenshot 2022-12-06 at 15 55 39" src="https://user-images.githubusercontent.com/112586815/205961782-0c806309-6737-43e0-86e4-6457ef7c5358.png">

rdig commented:
@jakubcolony this is not @danbr 's fault, it's because `colonyJS` doesn't support colony v11 yet

Apply this patch, restart your dev environement (the network docker image will re-build) and everything will start working again.
```diff
diff --git a/docker/colony-cdapp-dev-env-network b/docker/colony-cdapp-dev-env-network
index 3222ab2..2b5a2d2 100644
--- a/docker/colony-cdapp-dev-env-network
+++ b/docker/colony-cdapp-dev-env-network
@@ -1,7 +1,7 @@
-ARG NETWORK_HASH=develop
-
 FROM colony-cdapp-dev-env/base:latest
 
+ARG NETWORK_HASH=b4c34d3982fd3c1bf0e32a421cdfe8e24c665e7f
+
 # Declare volumes to set up metadata
 VOLUME [ "/colonyCDapp/amplify/mock-data" ]
 ```

This patch basically pins the colonyNetwork repo to a specific commit hash, right before they released colony v11

danbr commented:
> From what I was able to test, loading looks a bit ugly (I guess not a big deal since we've got a UI refresh coming): <img alt="Screenshot 2022-12-06 at 15 55 39" width="203" src="https://user-images.githubusercontent.com/112586815/205961782-0c806309-6737-43e0-86e4-6457ef7c5358.png">

I didn't touch styling in this PR. I didn't want to spend time on it as the UI refresh is on its way......

jakubcolony commented:
@rdig still getting the same error after changing the hash, I also tried the one from #74 (`e2194001e88dc129213a626a7d38943cbb0c9ef1`) but it didn't work either

chinins commented on amplify/backend/api/colonycdapp/schema.graphql:
```diff
@@ -130,28 +147,34 @@ enum ColonyType {
 
+enum SortingMethod {
+  BY_HIGHEST_REP
+  BY_LOWEST_REP
```
I know that this comes from the dapp but now it seems that we have the camel case for enums so I would update

jakubcolony replied:
We're using CamelCase for names but the members still seem to be all caps - have a look at `TokenType`

chinins replied:
Well, then we are not consistent with the FE, because I checked there and it's camel case. I don't really mind either way but I think it would be great to have consistency across everything.

chinins reviewed (commented):
It's all good, actually working fine for me - but i've been also using this branch in my own PR.

The only thing is that there is a little bug. When you have reputation in subdomain and you query this subdomain, no contributors are returned.

chinins reviewed (approved):
Working properly now.

willm30 reviewed (commented):
For me the patch works and the watchers / contributors show up in the correct section.

However, I don't think there should be a failure message when there are 0 watchers / contributors? I would expect something like `This colony has no contributors (/watchers)`. (I can't remember what we used in the dapp.)

![failedto](https://user-images.githubusercontent.com/64402732/207664914-f991fc73-96e2-45fd-9aa9-efb490a4cf5d.png)
![failed](https://user-images.githubusercontent.com/64402732/207664927-04249ebd-d8f3-496b-8ea8-900f1830dbc8.png)

I'm also seeing the same popover for each different user (User BORIS).
![allsame](https://user-images.githubusercontent.com/64402732/207666189-1b30fa47-cf19-48f7-accc-57a73abe3650.gif)

danbr commented:
> For me the patch works and the watchers / contributors show up in the correct section.
> 
> However, I don't think there should be a failure message when there are 0 watchers / contributors? I would expect something like `This colony has no contributors (/watchers)`. (I can't remember what we used in the dapp.)
> 

Agreed. I've updated to show zero (0) or the fetch error.

<img width="243" alt="Screenshot 2022-12-15 at 22 26 47" src="https://user-images.githubusercontent.com/582700/207900232-8a146580-2cf5-41a7-ad35-46d2dcc29e6e.png">


>I'm also seeing the same popover for each different user (User BORIS).

This was a bug with the wrong user being passed.
I've corrected, although in the screen captures the user name is not displayed. This is due to the User->Profile data being empty. UserInfoPopover.UserInfo is currently using displayName (in Profile). 

<img width="571" alt="Screenshot 2022-12-15 at 22 23 46" src="https://user-images.githubusercontent.com/582700/207899661-13e47994-4d02-4d9a-a9e6-24a61fba6349.png">
<img width="542" alt="Screenshot 2022-12-15 at 22 23 55" src="https://user-images.githubusercontent.com/582700/207899698-f8317f36-80a4-40a9-a3a6-45848afc1807.png">


and if i force to use User.name
<img width="609" alt="Screenshot 2022-12-15 at 22 11 07" src="https://user-images.githubusercontent.com/582700/207899141-7ce1e28d-d02f-49ad-b89e-f52ad8f0327f.png">
<img width="557" alt="Screenshot 2022-12-15 at 22 11 13" src="https://user-images.githubusercontent.com/582700/207899182-56e45bdd-f9a5-4eec-b212-9eeadaa86dde.png">

willm30 commented on amplify/backend/function/getMembersForColony/src/graphql.js:
```diff
@@ -0,0 +1,47 @@
+                  location
+                  thumbnail
+                  website
+                }
```
Query for `profileId` here as well to fix the problem with profile being `null` when consumed by the front end.

danbr replied:
@willm30 Great thanks, its working well now.

willm30 reviewed (approved):
Nice one. 

You can fix the problem with the profile showing up as `null` in the front end by adding `profileId` to your query in the back end. (I have left a comment)

![userLiz](https://user-images.githubusercontent.com/64402732/207950742-8c2112f4-e135-488f-acb4-f0b725747a23.png)

Pre-approving 👍
